libfwupdplugin/fu-context.c

```diff
@@ -1367,6 +1368,38 @@ fu_context_add_esp_volume(FuContext *self, FuVolume *volume)
 	g_ptr_array_add(priv->esp_volumes, g_object_ref(volume));
 }
 
+/**
+ * fu_context_set_preferred_esp_path:
+ * @self: The FuContext object.
+ * @path: The path to the preferred ESP.
+ * Sets the preferred ESP (EFI System Partition) path for the given FuContext.
+ */
+void
+fu_context_set_preferred_esp_path(FuContext *self, const gchar *path)
```

Comment:
I think `s\preferred_esp_path\esp_location` personally; it's what we're setting in the conf file.

Reply:
Then it prompts the question - what do we do if if doesn't match any of the paths _found_?

I picked preferred so it would still mostly work in that case.

Reply:
> what do we do if if doesn't match any of the paths found

I think that's actually OK. Imagine setting this sitewide, saying "prefer the first ESP" -- if you don't have two ESPs or your layout is very different we just ignore the config. Maybe we could emit a g_info or g_warning in that case.